test/validation/api/dma/dma.c

```diff
@@ -1320,7 +1320,7 @@ static void get_seg_lens(uint32_t max_len, uint32_t *src, uint32_t *dst)
 {
 	uint32_t src_segs = *src, dst_segs = *dst;
 
-	*src = max_len / src_segs;
+	*src = MIN(max_len / src_segs, global.dma_capa.max_seg_len);
 	*dst = *src * src_segs / dst_segs + *src * src_segs % dst_segs;
```

Comment:
Good catch, though I believe this might cause failures in case `max_src_segs` and `max_dst_segs` capabilities differ as the total data amount between source and destination segments could end up being different (if destination length was larger than maximum and needed truncation as well). Untested, but this could work:
```diff
@@ -1318,8 +1318,9 @@ static void test_dma_addr_to_addr_sync_res(void)
 
 static void get_seg_lens(uint32_t max_len, uint32_t *src, uint32_t *dst)
 {
-       uint32_t src_segs = *src, dst_segs = *dst;
+       uint32_t src_segs = *src, dst_segs = *dst, denom = MIN(src_segs, dst_segs);
 
+       max_len = MIN(max_len / denom, global.dma_capa.max_seg_len) * denom;
        *src = max_len / src_segs;
        *dst = *src * src_segs / dst_segs + *src * src_segs % dst_segs;
 }
```

Reply:
Thanks, I will spin up the next version with the above changes.